public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,885 @@
+ 
+  The component must copy the hero values in the _data model_ into the _form model_.
+  There are two important implications:
+
+  1. The developer must understand how the properties of the _data model_ 
+  map to the properties of the _form model_.
+
+  2. User changes flow from the DOM elements to the _form model_, not to the _data model_.
+  The form controls never update the _data model_.
+  In the reactive paradigm, the data model is presumed to be _immutable_
+  and it's your responsibility to preserve that immutability constraint.
+  Kara, Deborah suggested running this by you. Could you confirm that this is the case?
+
+  The _form_ and _data_ model structures need not match exactly.
+  You often present a subset of the _data model_ on a particular screen.
+  But it makes things easier if the shape of the _form model_ is close to the shape of the _data model_.
+  
+  In this `HeroDetailComponent`, the two models are quite close.
+
+  Recall the definition of `Hero` in `data-model.ts`:
++makeExample('reactive-forms/ts/app/data-model.ts', 'model-classes','app/data-model.ts (classes)')(format=".")
+:marked
+  Here, again, is the component's `FormGroup` definition.
+
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'hero-form-model','app/hero-detail.component.ts (excerpt)')(format=".")
+:marked
+  There are two significant differences between these models:
+
+  1. The `Hero` has an `id`. The form model does not because you generally don't show primary keys to users.
+
+  1. The `Hero` has an array of addresses. This form model presents only one address,
+  a choice [revisited below](#form-array "Form arrays").
+
+  Nonetheless, the two models are pretty close in shape and you'll see in a moment how this alignment facilitates copying the _data model_ properties
+  to the _form model_ with the `patchValue` and `setValue` methods.
+
+:marked
+  Take a moment to refactor the _address_ `FormGroup` definition for brevity and clarity as follows:
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'address-form-group')(format=".")
+:marked
+
+.l-main-section
+a#set-data
+:marked
+  ## Populate the form model with _setValue_ and _patchValue_
+  You can initialize control values in a `FormGroup` using `setValue` or `patchValue`.
+
+  ### _setValue_
+  With **`setValue`**, you assign _every_ form control value _at once_
+  by passing in a data object whose properties exactly match the _form model_ behind the `FormGroup`.
+
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'set-value','app/hero-detail.component.ts (excerpt)')(format=".")
+:marked
+  The `setValue` method checks the data object thoroughly before assigning any form control values.
+  It rejects a data object that doesn't match the `FormGroup` structure.
+  It rejects a data object if it is missing values for any control in the group.
+  It does return helpful errors.
```

Comment:
How about ...

The `setValue` method checks the data object thoroughly before assigning any form control values.
 * It rejects a data object that doesn't match the `FormGroup` structure.
 * It rejects a data object if it is missing values for any control in the group.
 * It returns helpful error messages.

If you have a typo or nested controls incorrectly, `setValue` will catch the error and report it clearly. [patchValue](#patchValue) will fail silently.